### PR TITLE
[RPC] Use apiConfiguration to limit gasPrice in eth_getGasPrice

### PR DESCRIPTION
**Pull request by Gabriel-Trintinalia**
## PR description
Use apiConfiguration to limit gasPrice in eth_getGasPrice

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthFeeHistory.java [hunk 1]
```diff
@@ -279,19 +283,9 @@ private List<Wei> boundRewards(final List<Wei> rewards) {
...
-    // If the reward is greater than the upper bound, return the upper bound.
-    if (reward.compareTo(upperBound) > 0) {
-      return upperBound;
-    }
-
-    // If the reward is within the bounds, return the reward as is.
-    return reward;
+    return reward.compareTo(lowerBound) <= 0
+        ? lowerBound
+        : reward.compareTo(upperBound) > 0 ? upperBound : reward;
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthFeeHistory.java:288]:
```suggestion
        : (reward.compareTo(upperBound) > 0) ? upperBound : reward;
```

Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthFeeHistory.java:288]: nit: this same logic in EthGasPrice has an extra set of () - idk which is easier to read but they should match

Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthFeeHistory.java:288]: another nit is that line 286 uses <= in the comparison but 288 uses strictly > but >= would have the same effect and would be more symmetrical

**Review by macfarla — APPROVED** [review]
couple of nits, otherwise LGTM

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthFeeHistory.java (hunk 1 above)
Comment (Gabriel-Trintinalia) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthFeeHistory.java]: done

**Pull request merged**